src/audio/eq_fir.c

```diff
@@ -63,20 +63,128 @@ struct comp_data {
 	uint32_t period_bytes;
 	int32_t *fir_delay;
 	size_t fir_delay_size;
+	void (*eq_fir_func_even)(struct fir_state_32x16 fir[],
+				 struct comp_buffer *source,
+				 struct comp_buffer *sink,
+				 int frames, int nch);
 	void (*eq_fir_func)(struct fir_state_32x16 fir[],
 			    struct comp_buffer *source,
 			    struct comp_buffer *sink,
 			    int frames, int nch);
-	void (*eq_fir_func_odd)(struct fir_state_32x16 fir[],
-				struct comp_buffer *source,
-				struct comp_buffer *sink,
-				int frames, int nch);
 };
 
-static void eq_fir_passthrough(struct fir_state_32x16 fir[],
-			       struct comp_buffer *source,
-			       struct comp_buffer *sink,
-			       int frames, int nch)
+/* The optimized FIR functions variants need to be updated into function
+ * set_fir_func. The cd->eq_fir_func is a function that can process any
+ * number of samples. The cd->eq_fir_func_even is for optimized version
+ * that is guaranteed to be called with even samples number.
+ */
+#if FIR_HIFI3
+static inline int set_fir_func(struct comp_dev *dev)
```

Comment:
Sorry I mean have one switch statement, but have multiple conditionally compiled versions of
```
static inline void set_s16_fir(struct comp_data *cd)
{
    cd->eq_func_even = blah;
    cd->eq->func = blah;
}
```
So the single switch just calls the static inline which is conditionally compiled.
```
static inline int set_fir_func(struct comp_dev *dev)
{
	struct comp_data *cd = comp_get_drvdata(dev);
	struct sof_ipc_comp_config *config = COMP_GET_CONFIG(dev);
 	switch (config->frame_fmt) {
	case SOF_IPC_FRAME_S16_LE:
		trace_eq("f16");
		set_s16_fir(cd);
		break;
	case SOF_IPC_FRAME_S24_4LE:
		trace_eq("f24");
		set_s24_fir(cd);
		break;
	case SOF_IPC_FRAME_S32_LE:
```